### PR TITLE
add terminating and ingress gateways

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- Add Vault TLS support for ingress and terminating gateways.

How I've tested this PR:
bats and acceptance tests added

How I expect reviewers to test this PR:
review tests

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/ingress-gateways-deployment.yaml [hunk 1]
```diff
@@ -55,6 +55,17 @@ spec:
         component: ingress-gateway
         ingress-gateway-name: {{ template "consul.fullname" $root }}-{{ .name }}
       annotations:
+        {{- if (and $root.Values.global.secretsBackend.vault.enabled $root.Values.global.tls.enabled) }}
+        "vault.hashicorp.com/agent-init-first": "true"
+        "vault.hashicorp.com/agent-inject": "true"
+        "vault.hashicorp.com/role": {{ $root.Values.global.secretsBackend.vault.consulCARole }}
```
Comment (t-eckert) [inline charts/consul/templates/ingress-gateways-deployment.yaml:61]: I'm always going to read this as someone yelling the first part of the name "Carol". "CAR-ole!"

Comment (kschoche) [reply charts/consul/templates/ingress-gateways-deployment.yaml:61]: hahaha

**Review by t-eckert — APPROVED** [review]
Great work on some beefy Bats tests. You'll have to give a seminar on writing bats.

**Comment by kschoche** [comment]
> Great work on some beefy Bats tests. You'll have to give a seminar on writing bats.

Thanks! I actually lifted them off https://github.com/hashicorp/consul-k8s/pull/889 from @ishustava !

File: acceptance/tests/vault/vault_test.go [hunk 2]
```diff
@@ -164,6 +165,11 @@ func TestVault(t *testing.T) {
 		"global.gossipEncryption.secretName": "consul/data/secret/gossip",
 		"global.gossipEncryption.secretKey":  "gossip",
 
+		"ingressGateways.enabled":               "true",
+		"ingressGateways.defaults.replicas":     "1",
+		"terminatingGateways.enabled":           "1",
```
Comment (ishustava) [inline acceptance/tests/vault/vault_test.go:170]:
```suggestion
		"terminatingGateways.enabled":           "true",
```

**Review by ishustava — APPROVED** [review]
🎉

File: acceptance/tests/vault/vault_test.go (hunk 2 above)
Comment (kschoche) [reply acceptance/tests/vault/vault_test.go:198]: copy/pasta spotting legend! thank you :)

**Pull request merged**